lldb/tools/lldb-dap/src-ts/debug-adapter-factory.ts

```diff
@@ -28,6 +23,70 @@ export class LLDBDapDescriptorFactory
     return true;
   }
 
+  static async findDAPExecutable(): Promise<string | undefined> {
+    let executable = "lldb-dap";
+    if (process.platform === "win32") {
+      executable = "lldb-dap.exe";
+    }
+
+    // Prefer lldb-dap from Xcode on Darwin.
+    if (process.platform === "darwin") {
+      try {
+        const exec = util.promisify(require("child_process").execFile);
```

Comment:
don't use require. Just import child process in the regular way, and then pass it to promisify. That way the typechecker will work correctly.